File: LeetCode/python3/942.py
class Solution:
    def diStringMatch(self, S: str) -> List[int]:
        S += 'E'
        l, h, n = 0, len(S) - 1, len(S)
        res = [0 for _ in range(n)]
        for i in range(n):
            if S[i] == 'I':
                res[i] = l
                l += 1
            else:
                res[i] = h
                h -= 1
        return res

File: LeetCode/cpp/942.cpp
class Solution {
public:
    vector<int> diStringMatch(string S) {
        S += 'E';
        int n = S.size(), m = S.size() - 1, j = 0;
        vector<int> ret(n, 0);
        for (int i = 0; i < n; ++i)
            ret[i] = S[i] == 'I' ? j++ : m--;
        return ret;
    }
};
